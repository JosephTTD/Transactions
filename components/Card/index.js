import React from "react";
import Container from "./style";

function Card({ amount, name, doe }) {
  const random = "8644765040641004";
  const cardno = random.split("");
  const show = cardno.filter((no, idx) => idx > 11);
  const encrypt = cardno.slice(-12, cardno.length).join("").replace(/./g, "*");
  const divide = encrypt.match(/.{1,3}/g).join(" ");

  return (
    <Container>
      <img src="images/mastercard.png" />
      <h3>{`Â£${amount || 0}`}</h3>
      <span>
        {divide} {show}
      </span>

      <div>
        {name && <p>{name}</p>}
        {doe && <p>{doe}</p>}
      </div>
    </Container>
  );
}

export default Card;
