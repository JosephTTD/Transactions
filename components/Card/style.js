import styled from "styled-components";

const Container = styled.div`
  background-image: url('images/card-bg.png');
  background-repeat: no-repeat;
  background-size: 110%;
  background-position: -20px;
  border-radius: 15px;
  color: black;
  height: 200px;
  margin-bottom: 20px;
  box-sizing: border-box;
  position: relative;
  padding: 25px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 auto;
  box-shadow: 0px 0px 20px 0px #eeeeee;

  h3 {
    font-size: 23px;
  } 

  div {
    position: absolute;
    display: flex;
    bottom: 0;
    justify-content: space-between;
    width: 100%;
    right: 0;
    padding-left: inherit;

    p {
        padding: 20px;
        margin: 0;
        font-size: 10px;

        &:first-child {
            padding-left: inherit;
        }

        &:nth-child(2) {
            padding-right: 30px;
            color: white;
        }
    }
  }

  img {
    width: 40px;
    position: absolute;
    right: 0;
    padding-right: inherit;
    padding-top: inherit;
    top: 0;
  }
`;

export default Container;
