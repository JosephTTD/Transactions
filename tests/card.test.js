import React from "react";
import * as renderer from "react-test-renderer";
import Card from "../components/Card";

/*jest.mock(Card, () => ({
  Card: (props) => {
    const MockCard = "mock-card";
    return <MockCard {...props} />;
  },
}));*/

const component = renderer.create(
  <Card amount="100" name="Jessy Day" doe="01/2023" />
);

describe("Card", () => {
  // snapshot tests
  it("renders correctly", () => {
    const component = renderer.create(<Card />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should render the prop values correctly", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  // write unit tests for Card component
  it("should render the value passed to the prop correctly on ", () => {
    const amount = component.root.findByType("h3").children[0];
    expect(amount).toBe("Â£100");
  });

  it("should render the name passed to the prop correctly", () => {
    const name = component.root.props.name;
    expect(name).toBe("Jessy Day");
  });

  it("should render the doe passed to the prop correctly", () => {
    const doe = component.root.props.doe;
    expect(doe).toBe("01/2023");
  });
});
