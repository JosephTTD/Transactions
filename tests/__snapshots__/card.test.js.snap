// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="sc-bczRLJ dqnLyA"
>
  <img
    src="images/mastercard.png"
  />
  <h3>
    £undefined
  </h3>
  <span>
    *** *** *** ***
     
    1
    0
    0
    4
  </span>
  <div />
</div>
`;

exports[`Card should render the prop values correctly 1`] = `
<div
  className="sc-bczRLJ dqnLyA"
>
  <img
    src="images/mastercard.png"
  />
  <h3>
    £100
  </h3>
  <span>
    *** *** *** ***
     
    1
    0
    0
    4
  </span>
  <div>
    <p>
      Jessy Day
    </p>
    <p>
      01/2023
    </p>
  </div>
</div>
`;
